velox/functions/lib/window/Ntile.cpp

```diff
@@ -216,22 +218,23 @@ class NtileFunction : public exec::WindowFunction {
 
   // Vector used to read the bucket column values.
   VectorPtr bucketVector_;
-  FlatVector<int64_t>* bucketFlatVector_;
+  FlatVector<TResult>* bucketFlatVector_;
 
   static const std::string kBucketErrorString;
 };
 
-const std::string NtileFunction::kBucketErrorString =
+template <typename TResult>
+const std::string NtileFunction<TResult>::kBucketErrorString =
     "Buckets must be greater than 0";
 
 } // namespace
 
-void registerNtile(const std::string& name) {
-  // ntile(bigint) -> bigint.
+template <typename TResult>
```

Comment:
Is this template parameter needed? Seems not in use. Thanks.

Reply:
@rui-mo we need to pass `TResult ` into NtileFunction.